src/App.js

```diff
@@ -1,19 +1,178 @@
+      this.setState({
+        containsWords: false,
+      });
+    } else {
+      this.setState({
+        containsWords: true,
+      });
+    }
+
+    this.setState({
+      todo: e.target.value,
+      inValidLength: false,
+    });
+  }
+
+  clickOnButton(todo, user, listOfTodos, isTooLong,
+    containsSymbols, containsWords) {
+    if (user.length > 0 && todo.length > 0 && !isTooLong
+      && !containsSymbols && containsWords) {
+      const targetUser = listOfTodos.find(person => (
+        person.name === user
+      ));
+
+      if (targetUser) {
+        targetUser.todo.push({
+          userId: targetUser.id,
+          id: Math.floor(Math.random() * 100),
+          title: todo,
+          completed: false,
+        });
+      }
+
+      this.forceUpdate();
+    }
+
+    if (user.length === 0) {
+      this.setState({
+        inValidSelect: true,
+      });
+    }
+
+    if (todo.length === 0) {
+      this.setState({
+        inValidLength: true,
+      });
+    }
+  }
+
+  render() {
+    const { listOfTodos, user, todo, inValidSelect, inValidLength,
+      isTooLong, containsSymbols, containsWords } = this.state;
+
+    return (
+      <div className="App">
+        <div>
+          <h1>Add todo form</h1>
+          <form
+            className="form"
+            method="POST"
+            action="/api/userTodo"
+            onSubmit={e => this.formSubmit(e, user, todo,
+              containsSymbols, isTooLong, containsWords)}
+          >
+            <select
+              className="form__field"
+              name="user"
+              value={user}
+              onChange={e => this.changeInSelect(e)}
+            >
+              <option>Choose a user</option>
+              {listOfTodos.map(person => (
+                <option
+                  key={person.id}
+                  value={person.name}
+                >
+                  {person.name}
+                </option>
+              ))}
+            </select>
+            {inValidSelect && <p>Please choose a user</p>}
+            <label>
+              <input
+                className="form__field form__field--wide"
+                name="todo"
+                placeholder="Add something to do"
+                value={todo}
+                onChange={e => this.changeInput(e)}
+              />
+            </label>
+            {inValidLength && <p>Please enter a Todo</p>}
+            {isTooLong
+              && <p>A Todo shouldn&apos;t be longer than 10 characters</p>}
+            {containsSymbols
+              && <p>A Todo should contain only latin characters or digits</p>}
+            {!containsWords && <p>Please, enter the words</p>}
+            <button
+              type="submit"
+              className="form__field"
+              onClick={() => (
+                this.clickOnButton(todo, user, listOfTodos,
+                  isTooLong, containsSymbols, containsWords)
+              )}
```

Comment:
you can get these values from state inside function, no need to pass them from jsx

```suggestion
              onClick={this.handleSubmit}
```